src/Order.java

```diff
@@ -65,19 +45,19 @@ public void printOrderDetails() {
         System.out.println("Date Shipped: " + dateShipped);
         System.out.println("User Name: " + userName);
         System.out.println("Order Status: " + orderStatus);
-        System.out.println("Shipping Address: " + shippingAddressLine1 + ", " + shippingAddressLine2 + ", " + shippingAddressCity + ", " + shippingAddressState + ", " + shippingAddressZip + ", " + shippingAddressCountry);
-        System.out.println("Billing Address: " + billingAddressLine1 + ", " + billingAddressLine2 + ", " + billingAddressCity + ", " + billingAddressState + ", " + billingAddressZip + ", " + billingAddressCountry);
+        System.out.println("Shipping Address: " + (shippingAddress != null ? shippingAddress : "N/A"));
+        System.out.println("Billing Address: " + (billingAddress != null ? billingAddress : "N/A"));
         System.out.println("Order Price: $" + orderPrice);
     }
 
-    public double calculatePrice(User user) {
-        double totalPrice = 0.0;
-
+    private float calculatePrice(User user) {
+        float totalPrice = 0f;
+    
         for (CartItem item : items) {
             totalPrice += item.getTotalPrice();
         }
-            totalPrice *= user.getDiscountRate(); 
-        
-        return totalPrice;
+    
+        return totalPrice * (float) user.getDiscountRate();
```

Comment:
Rather than casting to `float` here could you update the `getDiscountRate()` method in the `User.java` class so it returns a float? 
The `discountRate` field  in `User` should also be of type `float` and not `double`.
```suggestion
        return totalPrice * user.getDiscountRate();
```